components/server/ee/src/billing/billing-mode.ts

```diff
@@ -0,0 +1,174 @@
+                return { mode: "chargebee", canUpgradeToUBB: true };
+            }
+
+            // The user has at least one paid personal subscription
+            return {
+                mode: "chargebee",
+            };
+        }
+
+        // Stripe: Active personal subsciption?
+        const customer = await this.stripeSvc.findCustomerByUserId(user.id);
+        if (customer) {
+            const subscription = await this.stripeSvc.findUncancelledSubscriptionByCustomer(customer.id);
+            if (subscription) {
+                return { mode: "usage-based" };
+            }
+        }
+
+        // 3. Check team memberships/plans
+        // UBB overrides wins if there is _any_. But if there is none, use the existing Chargebee subscription.
+        const teamsModes = await Promise.all(teams.map((t) => this.getBillingModeForTeam(t, now)));
+        const hasUbbTeam = teamsModes.some((tm) => tm.mode === "usage-based");
+        const hasCbTeam = teamsModes.some((tm) => tm.mode === "chargebee");
+        const hasCbTeamSeat = cbTeamSubscriptions.length > 0;
+
+        if (hasUbbTeam) {
+            return { mode: "usage-based" }; // UBB is gready: once a user has at least a team seat, they should benefit from it!
+        }
+        if (hasCbTeam || hasCbTeamSeat) {
+            // TODO(gpl): Q: How to test the free-tier, then? A: Make sure you have no CB seats anymore
+            // For that we could add a new field here, which lists all seats that are "blocking" you, and display them in the UI somewhere.
+            return { mode: "chargebee", canUpgradeToUBB: true }; // UBB is enabled, but no seat nor subscription yet.
+        }
+
+        // UBB free tier
+        return { mode: "usage-based" };
+    }
+
+    async getBillingModeForTeam(team: Team, _now: Date): Promise<BillingMode> {
+        if (!this.config.enablePayment) {
+            // Payment is not enabled. E.g. Self-Hosted.
+            return { mode: "none" };
+        }
+        const now = _now.toISOString();
+
+        // Is Usage Based Billing enabled for this team?
+        const isUsageBasedBillingEnabled = await this.configCatClientFactory().getValueAsync(
+            "isUsageBasedBillingEnabled",
+            false,
+            {
+                teamId: team.id,
+                teamName: team.name,
+            },
+        );
+
+        // 1. UBB enabled?
+        if (!isUsageBasedBillingEnabled) {
+            return { mode: "chargebee" };
+        }
+
+        // 2. Any Chargbee TeamSubscription2 (old Team Subscriptions are not relevant here, as they are not associated with a team)
+        const teamSubscription = await this.teamSubscription2Db.findForTeam(team.id, now);
+        if (teamSubscription && TeamSubscription2.isActive(teamSubscription, now)) {
+            if (isUsageBasedBillingEnabled && TeamSubscription2.isCancelled(teamSubscription, now)) {
```

Comment:
Nit: I think you don't need to check `isUsageBasedBillingEnabled` here (we know it's `true`, because otherwise we would have already bailed out in the UBB check above)